docs/docs/recipes/integrate-logto/spring-boot.mdx

```diff
@@ -0,0 +1,226 @@
+---
+sidebar_label: Spring Boot
+---
+
+import redirectUriFigure from './assets/spring-boot-redirect-uri.png';
+import AppNote from './fragments/_app-note.mdx';
+import ConfigureRedirectUri from './fragments/_configure-redirect-uri.mdx';
+
+# Spring Boot
+
+<AppNote type="Traditional web" />
+
+Your web app may run on the server-side using Spring Boot framework.
+For now, you need to integrate Logto in Spring Boot manually.
+This article guides you on how to finish it step by step.
+And we use Java, Gradle, Spring MVC and [Thymeleaf](https://www.thymeleaf.org/) to take the example.
+
+## Start a Spring Boot project
+
+With [Spring Initializr](https://start.spring.io/), you can quickly start a Spring Boot project.
+Follow the options below:
+
+- Gradle Project
+- Language: Java
+- Spring Boot: 2.7.2
+
+Generate and open the project.
+
+## Add dependencies
+
+Add the dependencies to your Gradle project build file `build.gradle`:
+
+```groovy
+// path/to/project/build.gradle
+dependencies {
+    implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310")
+    implementation("org.springframework.boot:spring-boot-starter-web")
+    implementation("org.springframework.boot:spring-boot-starter-security")
+    implementation("org.springframework.boot:spring-boot-starter-oauth2-client")
+    implementation("org.springframework.boot:spring-boot-starter-thymeleaf")
+    implementation("org.thymeleaf.extras:thymeleaf-extras-springsecurity5")
+    implementation("nz.net.ultraq.thymeleaf:thymeleaf-layout-dialect")
+}
+```
+
+:::note
+
+Since Spring Boot and Spring Security have built-in support for both OAuth 2 and OIDC,
+you DO NOT need to add additional libraries from Logto to integrate.
+
+See [Spring Security and OpenID Connect](https://www.baeldung.com/spring-security-openid-connect)
+and [Spring Security Architecture](https://spring.io/guides/topicals/spring-security-architecture)
+for more details.
+
+:::
+
+## Configure Spring Security
+
+Spring Security makes it easy to configure your application for authentication with OIDC providers such as Logto.
+
+Use an `application.yml` file (instead of the default `application.properties`) to configure the OAuth2 client and provider.
+
+```yaml
+# path/to/project/src/main/resources/application.yaml
+spring:
+  security:
+    oauth2:
+      client:
+        registration:
+          logto:
+            client-id: <your-app-id>
+            client-secret: <your-app-secret> # Copy from application details in Admin Console
+            scope:
+              - openid
+              - profile
```

Comment:
```suggestion
              - profile
              - offline_access
```

Reply:
Since we add `offline_access` in scopes, ensure the sign-in URI contains `prompt=consent` and the fetched token response contains a refresh token.